test-tap/assert.js

```diff
@@ -120,8 +128,7 @@ function eventuallyFailsWith(t, fn, subset) {
 
 function fails(t, fn) {
 	lastFailure = null;
-	fn();
-	if (lastFailure) {
+	if (!fn() && lastFailure) {
```

Comment:
This should check more explicitly for a `false` value:
```suggestion
	if (fn() === false && lastFailure) {
```